File: mn-ping-java/src/main/java/bearmug/lambda/PingController.java
package bearmug.lambda;

import io.micronaut.http.annotation.Controller;
import io.micronaut.http.annotation.Get;
import reactor.core.publisher.Mono;

import javax.inject.Inject;

@Controller("/ping")
public class PingController {

	@Inject
    PingService pingService;

    @Get("/java")
    public String pingJava() {
        return pingService.pong();
    }

    @Get("/graal")
    public String pingGraal() {
        return pingService.pong();
    }

    @Get("/rjava")
    public Mono<String> pingJavaReactive() {
        return Mono.just(pingService.pong());
    }

    @Get("/rgraal")
    public Mono<String> pingGraalReactive() {
        return Mono.just(pingService.pong());
    }
}

File: mn-ping-kotlin/src/main/kotlin/bearmug/lambda/PingController.kt
package bearmug.lambda

import io.micronaut.http.annotation.Controller
import io.micronaut.http.annotation.Get
import reactor.core.publisher.Mono

@Controller("/ping")
class PingController(
        private val pingService: PingService) {

    @Get("/kotlin")
    fun pingKotlin() = pingService.pong()

    @Get("/graal")
    fun pingGraal() = pingService.pong()

    @Get("/rkotlin")
    fun pingKotlinReactive() = Mono.just(pingService.pong())

    @Get("/rgraal")
    fun pingGraalReactive() = Mono.just(pingService.pong())
}